codersdk/workspaceagents.go

```diff
@@ -285,8 +285,8 @@ func (c *Client) DialWorkspaceAgentTailnet(ctx context.Context, logger slog.Logg
 				return
 			}
 			if err != nil {
+				// WARN: closing here may lead to nhooyr websocket panicking.
```

Comment:
Good fix, although I would recommend rewriting the above error check as well. A simple `if err != nil` should suffice here since we should never access `ws` if `Dial` returns an error.